### ISSUE TITLE
v2 Playground: No type errors in JS mode

### ISSUE THREAD

**Issue by IllusionMH** [microsoft/TypeScript-Website#203]
Even [simple example](https://www.typescriptlang.org/play/index.html?useJavaScript=true#code/PQKhFgCgAIWgBADgQwE7ILbQN4DsCuGARgKaoC+0yUswUAZvrgMYAuAlgPa7QAmJGTgApkAShw1oU1CVb5UPAOSLoAaioBuKOShR+goa1T4SojdGDBouTtDKpOqIA) doesn't report errors even if checkJS option appears to be checked in Config dropdown.

```js
/**
 * @param {number} a
 */
function demo(a) {
    return '' + a;
}

demo(true); // no error
```

Is it expected behavior? I haven't found direct mention that type checking is expected not to work, however no examples where if should show errors.

UPD. v3 has all errors in Errors tab, but no inline which is #201. Can it be a same root cause here?

**Comment by orta** [comment]
Interesting, maybe https://github.com/microsoft/monaco-typescript/pull/40 isn't happening anymore in the Monaco-typescripts since moving to the TS CDN

**Comment by orta** [comment]
( The errors tab is working because that use the compiler API, and not monaco, so it would work regardless )

**Comment by orta** [comment]
Fixed

**Comment by IllusionMH** [comment]
Doesn't look fixed for v2

```js
/** @type {string} */
let b = 123;
```
Fixed [Playground v3](https://www.staging-typescript.org/play?ssl=1&ssc=1&pln=2&pc=13&useJavaScript=true#code/PQKhAIAEBcE8AcCm4DeBnaAnAlgOwOYC+4IwAUADaLTgBG4AvOAIwBMAzANxA)

No error [Playground v2](https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=2&pc=13&useJavaScript=true#code/PQKhAIAEBcE8AcCm4DeBnaAnAlgOwOYC+4IwAUADaLTgBG4AvOAIwBMAzANxA) (no errors Nightly)

![image](https://user-images.githubusercontent.com/1312662/87208927-e6617700-c318-11ea-8be1-7b21fb7f2c6e.png)

**Comment by orta** [comment]
Ah, I'll keep this open till it's completely deprecated then

**Comment by orta** [comment]
This is fixed I the current playground

**Comment by IllusionMH** [comment]
Awesome, thank you!
